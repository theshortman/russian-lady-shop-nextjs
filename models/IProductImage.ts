export interface IProductImage {
  id: number;
  image_large: string;
  image_medium: string;
  image_small: string;
}
