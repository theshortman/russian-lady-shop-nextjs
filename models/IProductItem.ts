export interface IProductItem {
  id: number;
  size: number;
  quantity: number;
}
