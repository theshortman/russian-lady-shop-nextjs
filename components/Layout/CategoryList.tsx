import { FC } from "react";
import { ICategory } from "@/models/ICategory";
import CategoryListItem from "@/components/Layout/CategoryListItem";

interface CategoryListProps {
  categories: ICategory[];
}

const CategoryList: FC<CategoryListProps> = ({ categories }) => {
  return (
    <div className="list-groupn mb-3">
      {categories?.map((category) => (
        <CategoryListItem key={category.id} {...category} />
      ))}
    </div>
  );
};

export default CategoryList;
