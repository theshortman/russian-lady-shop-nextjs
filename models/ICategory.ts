import { IProduct } from "@/models/IProduct";

export interface ICategory {
  id: number;
  name: string;
  description: string;
  slug: string;
  products: IProduct[];
  prev_page_number: number | null;
  next_page_number: number | null;
}
