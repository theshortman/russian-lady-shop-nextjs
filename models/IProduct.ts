import { IProductImage } from "@/models/IProductImage";
import { IProductItem } from "@/models/IProductItem";

export interface IProduct {
  id: number;
  name: string;
  slug: string;
  description: string;
  detail: string;
  price: number;
  discount: number;
  new_price: number;
  product_images: IProductImage[];
  product_items: IProductItem[];
}
