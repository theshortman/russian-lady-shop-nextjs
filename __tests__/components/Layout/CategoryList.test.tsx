import { render } from "@testing-library/react";
import CategoryList from "@/components/Layout/CategoryList";
import { ICategory } from "@/models/ICategory";

test("renders CategoryList unchanged", () => {
  const category: ICategory = {
    id: 1,
    name: "Test category",
    description: "Test description",
    slug: "test-slug",
    products: [],
    prev_page_number: 1,
    next_page_number: 2,
  };

  const { container } = render(<CategoryList categories={[category]} />);
  expect(container).toMatchSnapshot();
});
