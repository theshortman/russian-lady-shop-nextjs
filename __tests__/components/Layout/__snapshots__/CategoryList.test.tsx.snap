// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CategoryList unchanged 1`] = `
<div>
  <div
    class="list-groupn mb-3"
  >
    <a
      aria-current="page"
      class="list-group-item list-group-item-action"
      href="/catalog/test-slug"
    >
      Test category
    </a>
  </div>
</div>
`;
